src/swe/physics_swe.c

```diff
@@ -51,14 +51,15 @@ static PetscErrorCode InitMPITypesAndOps(void) {
 
   if (!initialized) {
     // create an MPI data type for the CourantNumberDiagnostics struct
-    const int      num_blocks             = 3;
-    const int      block_lengths[3]       = {1, 1, 1};
-    const MPI_Aint block_displacements[3] = {
+    const int      num_blocks             = 4;
+    const int      block_lengths[4]       = {1, 1, 1, 1};
+    const MPI_Aint block_displacements[4] = {
         offsetof(CourantNumberDiagnostics, max_courant_num),
         offsetof(CourantNumberDiagnostics, global_edge_id),
         offsetof(CourantNumberDiagnostics, global_cell_id),
+        offsetof(CourantNumberDiagnostics, is_set),
     };
-    MPI_Datatype block_types[3] = {MPI_DOUBLE, MPI_INT, MPI_INT};
+    MPI_Datatype block_types[4] = {MPI_DOUBLE, MPI_INT, MPI_INT, MPIU_BOOL};
```

Comment:
Jed suggested using `MPIU_REAL` insteadl of `MPI_DOUBLE` here. I think that handles both single and double precision (in case it ever matters).